src/signature.c

```diff
@@ -870,7 +870,7 @@ static CK_RV p11prov_sig_operate_init(P11PROV_SIG_CTX *sigctx, bool digest_op,
     }
 
 done:
-    if (ret != CKR_OK) {
+    if (ret != CKR_OK && ret != CKR_OBJECT_HANDLE_INVALID) {
```

Comment:
if you do the refreshing in this function, as you should, then this will go away, as it should.
This exception here is a trap for any of the callers.